components/data-bridge/org.wso2.carbon.databridge.streamdefn.filesystem/src/main/java/org/wso2/carbon/databridge/streamdefn/filesystem/internal/EventStreamListenerImpl.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.databridge.streamdefn.filesystem.internal;
```

Comment:
I think it's better to have EventStreamListener implementation in the databridge core where the AbstractStreamDefintionStore is existing. And then when the removedEventStream is called, we can call invalidateStreamDefinitionStore there. Moving this away from file system, and having it in the common place is better IMHO because later on if any implementing new stream definition store is not necessarily need to do this.